pkg/operator/controller/controller_dns_test.go

```diff
@@ -3,199 +3,111 @@ package controller
-	Spec: configv1.DNSSpec{
-		PublicZone: &publicZone,
-	},
-}
+	corev1 "k8s.io/api/core/v1"
 
-func TestDesiredDNSRecords(t *testing.T) {
-	type ingress struct {
-		host string
-		ip   string
-	}
-	type record struct {
-		typ    dns.RecordType
-		name   string
-		target string
-		zone   configv1.DNSZone
-	}
-	makeService := func(ingresses []ingress) *corev1.Service {
-		service := &corev1.Service{}
-		for _, ingress := range ingresses {
-			service.Status.LoadBalancer.Ingress = append(service.Status.LoadBalancer.Ingress, corev1.LoadBalancerIngress{
-				IP:       ingress.ip,
-				Hostname: ingress.host,
-			})
-		}
-		return service
-	}
-	makeRecords := func(records []record) []*dns.Record {
-		dnsRecords := []*dns.Record{}
-		for _, r := range records {
-			record := &dns.Record{
-				Zone: r.zone,
-				Type: r.typ,
-			}
-			switch record.Type {
-			case dns.ALIASRecord:
-				record.Alias = &dns.AliasRecord{
-					Domain: r.name,
-					Target: r.target,
-				}
-			case dns.ARecordType:
-				record.ARecord = &dns.ARecord{
-					Domain:  r.name,
-					Address: r.target,
-				}
-			}
-			dnsRecords = append(dnsRecords, record)
-		}
-		return dnsRecords
-	}
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestDesiredWildcardDNSRecord(t *testing.T) {
 	tests := []struct {
 		description string
 		domain      string
 		publish     operatorv1.EndpointPublishingStrategyType
-		dnsConfig   *configv1.DNS
-		ingresses   []ingress
-		expect      []record
+		ingresses   []corev1.LoadBalancerIngress
+		expect      *iov1.DNSRecordSpec
 	}{
 		{
 			description: "no domain",
 			domain:      "",
 			publish:     operatorv1.LoadBalancerServiceStrategyType,
-			dnsConfig:   globalConfig,
-			ingresses: []ingress{
-				{host: "lb.cloud.example.com", ip: ""},
-				{host: "lb.cloud.example.com", ip: "192.0.2.1"},
+			ingresses: []corev1.LoadBalancerIngress{
+				{Hostname: "lb.cloud.example.com"},
 			},
-			expect: []record{},
+			expect: nil,
 		},
 		{
 			description: "not a load balancer",
 			domain:      "apps.openshift.example.com",
 			publish:     operatorv1.HostNetworkStrategyType,
-			dnsConfig:   globalConfig,
-			expect:      []record{},
-		},
-		{
-			description: "no zones",
-			domain:      "apps.openshift.example.com",
-			publish:     operatorv1.LoadBalancerServiceStrategyType,
-			dnsConfig:   &configv1.DNS{},
-			ingresses: []ingress{
-				{host: "lb.cloud.example.com"},
-			},
-			expect: []record{},
-		},
-		{
-			description: "public only ALIAS",
-			publish:     operatorv1.LoadBalancerServiceStrategyType,
-			domain:      "apps.openshift.example.com",
-			dnsConfig:   publicConfig,
-			ingresses: []ingress{
-				{host: "lb.cloud.example.com"},
-			},
-			expect: []record{
-				{typ: dns.ALIASRecord, name: "*.apps.openshift.example.com", target: "lb.cloud.example.com", zone: publicZone},
-			},
+			expect:      nil,
 		},
 		{
-			description: "private only A",
+			description: "no ingresses",
+			domain:      "",
```

Comment:
This test case should have a non-empty value for `domain` because an empty value also causes a nil result.

Reply:
Fixed